### DISCUSSION TITLE
How does ArgoCD decide if an application should be updated or freshly installed?

### DISCUSSION THREAD

**Discussion by meysholdt** [argoproj/argo-cd#7622] [Q&A]
Hi there!

We have a cluster in which we install the same application several times. let's call the application A. Example:
* in namespace foo, A is installed via ArgoCD.
* in namespace bar, A is installed `kubectl apply`.
* in namespace baz, A is installed `kubectl apply`.

When we now run `argocd sync` on `foo.A`, ArgoCD wants to overwrite sometimes `bar.A` and sometimes `baz.A`.

Im surprised this happens, because we specified namespace `foo` as destination in the application manifest of A in ArgoCD.

Is this expected behaviour or am I missing something? 

Is it required to restrict `foo.A` to namespace `foo` via Cluster Manifest or Project manifest?

When `argocd sync` runs, what are the exact criteria to match k8s resources? I kinda expected it would be k8s-namespace and k8s-resourcename. But based on what we observed, the namespace does not seem to be applied for matching.

**Comment by hjorth** [comment]
https://argo-cd.readthedocs.io/en/release-1.8/faq/#why-is-my-app-out-of-sync-even-after-syncing

Any chance it could be related to this?
